### PR TITLE
Pyinstaller hook

### LINKED ISSUE #454: PyInstaller hooks needed
When bundling pygfx with PyInstaller into an application, there are some data files and some dynamic libs missing.

We need to bundle:

* dynamic libs of `freetype` (collect_dynamic_libs)
* data from `pygfx.pkg_resources` (collect_data_files)

I noticed that when just using `collecting_data_files("pygfx.pkg_resources")` results in a name clash with `pkg_resources` module from setuptools. We might consider replacing the use of `pkg_resources` with `importlib.resources`?

### PR DESCRIPTION
**Pull request by Korijn**
Closes #454

Note: I renamed `pkg_resources` to `data_files` and not `resources` because that is already an existing subpackage

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -87,6 +87,23 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+    - name: Install package and dev dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (ivoflipse) [inline .github/workflows/ci.yml:101]: Is the current pip outdated?

Comment (Korijn) [reply .github/workflows/ci.yml:101]: I just copy pasted this, will remove it

Comment (Korijn) [reply .github/workflows/ci.yml:101]: Actually, no, I'll keep it around. The other jobs use the command as well.

File: pygfx/__pyinstaller/test_pygfx.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+def test_pyi_pygfx(pyi_builder):
+    pyi_builder.test_source(
+        """
+        import pygfx
```
Comment (almarklein) [inline pygfx/__pyinstaller/test_pygfx.py:4]:
```suggestion
        import pygfx as gfx
        ff = gfx.font_manager.select_font("foo", gfx.font_manager.default_font_props)[0][1]
        print(ff.codepoints)
        assert len(ff.codepoints) > 100
```

**Review by almarklein — APPROVED** [review]

**Pull request merged**